src/Markup/Avalonia.Markup/Markup/Parsers/Reader.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Avalonia.Markup.Parsers
 {
-    internal class Reader
+    public class Reader
```

Comment:
@jkoritzinsky I made this class public so it could be used in Avalonia.Markup.Xaml for parsing property strings. Given that, should we rename it? Maybe `CharacterReader` or something? `Reader` feels a bit vague.

Reply:
In fact maybe it should also be moved into `Avalonia.Base\Utilities` alongside `StringTokenizer`?

Reply:
We could rename and move! That works for me.

Reply:
Done. I also moved `IdentifierParser`.